h/models/moderation_log.py

```diff
@@ -10,17 +12,22 @@
 from h.db.mixins_dataclasses import AutoincrementingIntegerID, CreatedMixin
 from h.models.annotation import ModerationStatus
 
+if TYPE_CHECKING:
+    from h.models import Annotation, User
+
 
 class ModerationLog(Base, AutoincrementingIntegerID, CreatedMixin, MappedAsDataclass):
     __tablename__ = "moderation_log"
 
-    moderator_id: Mapped[int] = mapped_column(ForeignKey("user.id", ondelete="CASCADE"))
-    moderator = relationship("User")
-
-    annotation_id: Mapped[types.URLSafeUUID] = mapped_column(
-        ForeignKey("annotation.id", ondelete="CASCADE"), index=True
-    )
-    annotation = relationship("Annotation")
+    moderator: Mapped[Optional["User"]] = relationship("User")
```

Comment:
Using Optional instead of `| None` as the later doesn't work with string references to types.